src/DataTypes/DataTypeEnum.cpp

```diff
@@ -191,9 +211,19 @@ void DataTypeEnum<Type>::serializeTextXML(const IColumn & column, size_t row_num
 template <typename Type>
 void DataTypeEnum<Type>::deserializeTextJSON(IColumn & column, ReadBuffer & istr, const FormatSettings &) const
 {
-    std::string field_name;
-    readJSONString(field_name, istr);
-    assert_cast<ColumnType &>(column).getData().push_back(getValue(StringRef(field_name)));
+    if (*istr.position() != '"')
```

Comment:
`istr.position()` can be at the end of the buffer. So, you should add check `!istr.eof()` before.